packages/node/src/channels.ts

```diff
@@ -407,46 +233,115 @@ export class Channels {
     return apps;
   }
 
+  private generateNewMultisigAddress(owners: Address[]): Address {
+    // TODO: implement this using CREATE2
+    return Wallet.createRandom().address;
+  }
+
   /**
-   * Gets all proposed appInstances across all of the channels open on
-   * this Node.
+   * @param appInstanceInfo The AppInstanceInfo to convert
+   * @param channel The channel the AppInstanceInfo belongs to
    */
-  private async getProposedAppInstances(): Promise<AppInstanceInfo[]> {
-    const apps: AppInstanceInfo[] = [];
-    const channels = await this.getAllChannelsJSON();
-    Object.values(channels).forEach((channel: Channel) => {
-      if (channel.proposedAppInstances) {
-        apps.push(...Object.values(channel.proposedAppInstances));
-      } else {
-        console.log(
-          `No app instances exist for channel with multisig address: ${
-            channel.multisigAddress
-          }`
-        );
-      }
-    });
-    return apps;
-  }
+  private createAppInstanceFromAppInstanceInfo(
+    proposedAppInstanceInfo: ProposedAppInstanceInfo,
+    channel: StateChannel
+  ): AppInstance {
+    const appInstanceInfo = { ...proposedAppInstanceInfo };
+    const appFunctionSigHashes = this.getAppFunctionSigHashes(appInstanceInfo);
+
+    const appInterface: AppInterface = {
+      addr: appInstanceInfo.appId,
+      applyAction: appFunctionSigHashes.applyAction,
+      resolve: appFunctionSigHashes.resolve,
+      getTurnTaker: appFunctionSigHashes.getTurnTaker,
+      isStateTerminal: appFunctionSigHashes.isStateTerminal,
+      stateEncoding: appInstanceInfo.abiEncodings.stateEncoding,
+      actionEncoding: appInstanceInfo.abiEncodings.actionEncoding
+    };
+
+    const terms: Terms = {
+      assetType: appInstanceInfo.asset.assetType,
+      limit: appInstanceInfo.myDeposit.add(appInstanceInfo.peerDeposit)
+    };
```

Comment:
This is not typed correctly... weird that linter did not complain.

```
Property 'token' is missing in type '{ assetType: AssetType; limit: BigNumber; }' but required in type 'Terms'.ts(2741)
```

Reply:
Oh I see you added `?` to `token` in the type.

I don't think it is a good idea to add `?`.

I would recommend just setting the value to `AddressZero` by default.